### PR TITLE
NNVP-86

### PR DESCRIPTION
**Pull request by eryk7b**

### CODE REVIEW

**Comment by theglue-cicd** [comment]
:star: PR built and available in a preview environment **the-glue-nnvp-backend-pr-44** [here](http://nnvp-backend.jx-the-glue-nnvp-backend-pr-44.cluster.development.nnvp.thegluesandbox.com)

File: src/main/java/com/theglue/nnvp/controller/VehicleSearchController.java [hunk 1]
```diff
@@ -145,6 +153,58 @@ public VehicleSearchController(MapperProvider mapperProvider,
...
+  private Vehicle getVehicleByVinOrLicensePlate(GetVehicleByVinOrLicensePlateRequestDTO request) {
+    // TODO NNVP-86:
+    //  - make sure it's synced with VehiclesByVinOrLicensePlateRequestValidator
+    if (StringUtils.isNotBlank(request.getVin())) {
+      return vehicleVinService.getByVin(request.getVin(), request.getReplyType());
+    }
+    if (StringUtils.isNotBlank(request.getLicensePlate())) {
+      return vehicleLicencePlateService.getByLicensePlate(request.getVin(), request.getReplyType());
+    }
+    throw new RuntimeException();
```
Comment (mateusztheglue) [inline src/main/java/com/theglue/nnvp/controller/VehicleSearchController.java]: that should be handled in validator

File: src/main/java/com/theglue/nnvp/controller/VehicleSearchController.java [hunk 2]
```diff
@@ -145,6 +153,58 @@ public VehicleSearchController(MapperProvider mapperProvider,
     return new ResponseEntity<>(SuccessResponseDTO.successResponse(response), HttpStatus.OK);
   }
```
Comment (mateusztheglue) [inline src/main/java/com/theglue/nnvp/controller/VehicleSearchController.java]: I think /vehicles endpoint is not used anymore now - in that case it should be removed

**Comment by theglue-cicd** [comment]
:star: PR built and available in a preview environment **the-glue-nnvp-backend-pr-44** [here](http://nnvp-backend.jx-the-glue-nnvp-backend-pr-44.cluster.development.nnvp.thegluesandbox.com)

**Pull request merged**